src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -32,7 +32,7 @@ private void InitSort(CultureInfo culture)
             }
             else
             {
-                _isAsciiEqualityOrdinal = (_sortName == "en-US" || _sortName == "");
+                _isAsciiEqualityOrdinal = _sortName == "" || (_sortName.Length >= 2 && _sortName[0] == 'e' && _sortName[1] == 'n');
```

Comment:
>sortName.Length >= 2 && _sortName[0] == 'e' && _sortName[1] == 'n') [](start = 63, length = 67)

you need to test for the dash too. 

You may just do sortName.StartsWith("en-", Ordinal)

Reply:
One more thing to validate is ensure the sortName is always normalized having the first language part is always lower case . I know it is but will be good to validate it.

Reply:
Can't call `StartsWith(..., Ordinal)` reliably because we're in the process of initializing the ordinal comparer. :)

But +1 to what was said earlier: test for "en" or "en-\*", not "en\*".

Reply:
> Can't call StartsWith(..., Ordinal) reliably because we're in the process of initializing the ordinal comparer. :)

We are inside culture aware things. I believe StartWith(... Ordinal) should work seamlessly as it'll never enter this initialization code.

Reply:
There's no concern that it'll cause reentrancy issues, such as trying to construct the Invariant culture's information since it's referenced by the StartsWith method?

Reply:
There is no such problem today, but it is best to avoid potential dependency cycles like these. Also, checking for the individual characters would be smaller/faster code.